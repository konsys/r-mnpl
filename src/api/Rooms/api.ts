import {
  IAddPlayerToRoom,
  IRoomResponce,
  IRoomState,
} from "../../stores/Game/Rooms/RoomsModel";

import { IResponceCode } from "types/types";
import { client } from "http/client";

const roomsUrl = `/rooms`;

export const fetchRooms = async (): Promise<IRoomResponce> => {
  return await (await client.get(roomsUrl)).data;
};

export const createRoomFetch = async (
  params: IRoomState
): Promise<IResponceCode> => {
  return await (await client.post(roomsUrl, { room: params })).data;
};

export const addPlayerToRoomFetch = async (
  params?: IAddPlayerToRoom
): Promise<IResponceCode> => {
  return await (await client.post(`${roomsUrl}/addPlayer`, { add: params }))
    .data;
};

export const removePlayerFromRoomFetch = async (
  params?: IAddPlayerToRoom
): Promise<IResponceCode> => {
  return await (
    await client.post(`${roomsUrl}/removePlayer`, { remove: params })
  ).data;
};

export const fetchRoom = async (gameId: string): Promise<IRoomState> => {
  return await (await client.get(`/rooms/${gameId}`)).data;
};
