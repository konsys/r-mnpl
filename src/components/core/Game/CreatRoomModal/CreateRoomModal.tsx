import "./style.scss";

import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  Grid,
  Typography,
} from "@material-ui/core";
import {
  RoomType,
  createRoom,
  newRoomStore,
  updateRoom,
} from "stores/Game/Room/NewRoomStore";
import {
  closeRoomModal,
  roomModalStore,
} from "stores/Game/Room/RoomModalStore";

import CloseIcon from "@material-ui/icons/Close";
import QuickGameParams from "./GameParams/QuickGameParams";
import React from "react";
import RegularGameParams from "./GameParams/RegularGameParams";
import RetroGameParams from "./GameParams/RetroGameParams";
import RouletteGameParams from "./GameParams/RouletteGameParams";
import ShuffleGameParams from "./GameParams/ShuffleGameParams";
import { useStore } from "effector-react";
import { useTranslation } from "react-i18next";

export default function CreateRoomModal() {
  const open = useStore(roomModalStore);
  const { t } = useTranslation();
  const room = useStore(newRoomStore);
  const { roomType } = room;
  return (
    <Dialog open={open} onClose={() => null}>
      <DialogContent className={"newRoom"}>
        <Grid container justify="flex-end" alignItems="center">
          <CloseIcon
            style={{ cursor: "pointer" }}
            onClick={() => closeRoomModal()}
          />
        </Grid>
        <Grid
          container
          justify="space-between"
          alignItems="flex-start"
          direction="row"
          spacing={1}
          className={"roomContent"}
        >
          <Grid item sm={5}>
            <Grid
              container
              direction="column"
              spacing={2}
              className="menuBlock"
            >
              <Grid
                item
                className={
                  roomType === RoomType.REGULAR
                    ? RoomType.REGULAR + " _selected"
                    : RoomType.REGULAR
                }
                onClick={() =>
                  updateRoom({ ...room, roomType: RoomType.REGULAR })
                }
              >
                <Typography variant="h6">{t("Regular game")}</Typography>
                <Typography variant="body2">
                  {t("Usual rating match.")}
                </Typography>
              </Grid>
              <Grid
                item
                className={
                  roomType === RoomType.QUICK
                    ? RoomType.QUICK + " _selected"
                    : RoomType.QUICK
                }
                onClick={() =>
                  updateRoom({ ...room, roomType: RoomType.QUICK })
                }
              >
                <Typography variant="h6">{t("Quick game")}</Typography>
                <Typography variant="body2">
                  {t("Fird dice and other rules for quick match.")}
                </Typography>
              </Grid>
              <Grid
                item
                className={
                  roomType === RoomType.SHUFFLE
                    ? RoomType.SHUFFLE + " _selected"
                    : RoomType.SHUFFLE
                }
                onClick={() =>
                  updateRoom({ ...room, roomType: RoomType.SHUFFLE })
                }
              >
                <Typography variant="h6">{t("GMS Shuffle")}</Typography>
                <Typography variant="body2">
                  {t("Tournament for playing with different cards.")}
                </Typography>
              </Grid>
              <Grid
                item
                className={
                  roomType === RoomType.RETRO
                    ? RoomType.RETRO + " _selected"
                    : RoomType.RETRO
                }
                onClick={() =>
                  updateRoom({ ...room, roomType: RoomType.RETRO })
                }
              >
                <Typography variant="h6">{t("Retro")}</Typography>
                <Typography variant="body2">
                  {t("Rules for classic monopoly")}
                </Typography>
              </Grid>
              <Grid
                item
                className={
                  roomType === RoomType.ROULETTE
                    ? RoomType.ROULETTE + " _selected"
                    : RoomType.ROULETTE
                }
                onClick={() =>
                  updateRoom({ ...room, roomType: RoomType.ROULETTE })
                }
              >
                <Typography variant="h6">{t("Russian roulette")}</Typography>
                <Typography variant="body2">
                  {t("Go and shoot. Nothing more.")}
                </Typography>
              </Grid>
            </Grid>
          </Grid>
          <Grid item sm={6} className="roomParams">
            <Grid container justify="space-between" alignItems="center">
              {roomType === RoomType.REGULAR && <RegularGameParams />}
              {roomType === RoomType.QUICK && <QuickGameParams />}
              {roomType === RoomType.RETRO && <RetroGameParams />}
              {roomType === RoomType.ROULETTE && <RouletteGameParams />}
              {roomType === RoomType.SHUFFLE && <ShuffleGameParams />}
            </Grid>
          </Grid>
        </Grid>
      </DialogContent>
      <DialogActions>
        <Button
          onClick={() => {
            createRoom();
            closeRoomModal();
          }}
          color="primary"
          variant="outlined"
        >
          {t("Create room")}
        </Button>
      </DialogActions>
    </Dialog>
  );
}
