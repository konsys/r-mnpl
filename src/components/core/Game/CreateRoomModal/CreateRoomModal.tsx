import "./style.scss";

import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  Grid,
  Typography,
} from "@material-ui/core";
import {
  RoomType,
  createRoom,
  currentRoom$,
  updateRoom,
} from "stores/Game/Rooms/RoomsModel";
import {
  closeRoomModal,
  roomModalStore,
} from "stores/Game/Rooms/RoomsModalModel";

import CloseIcon from "@material-ui/icons/Close";
import QuickGameParams from "./GameParams/QuickGameParams";
import React from "react";
import RegularGameParams from "./GameParams/RegularGameParams";
import RetroGameParams from "./GameParams/RetroGameParams";
import RouletteGameParams from "./GameParams/RouletteGameParams";
import ShuffleGameParams from "./GameParams/ShuffleGameParams";
import { useStore } from "effector-react";
import { useTranslation } from "react-i18next";
import { user$ } from "stores/Game/User/UserModel";

export default function CreateRoomModal() {
  const user = useStore(user$);
  const open = useStore(roomModalStore);
  const { t } = useTranslation();
  const room = useStore(currentRoom$);
  const { roomType } = room;
  return (
    <Dialog open={open} onClose={() => null}>
      <DialogContent className={"newRoom"}>
        <Grid container justify="flex-end" alignItems="center">
          <CloseIcon
            style={{ cursor: "pointer" }}
            onClick={() => closeRoomModal()}
          />
        </Grid>
        <Grid
          container
          justify="space-between"
          alignItems="flex-start"
          direction="row"
          spacing={1}
          className={"roomContent"}
        >
          <Grid item sm={5}>
            <Grid
              container
              direction="column"
              spacing={2}
              className="menuBlock"
            >
              <Grid
                item
                className={
                  "regular " +
                  (roomType === RoomType.REGULAR
                    ? RoomType.REGULAR + " _selected"
                    : RoomType.REGULAR)
                }
                onClick={() =>
                  updateRoom({ ...room, roomType: RoomType.REGULAR })
                }
              >
                <Typography variant="h6">{t("Regular game")}</Typography>
                <Typography variant="body2">
                  {t("Usual rating match.")}
                </Typography>
              </Grid>
              <Grid
                item
                className={
                  "quick " +
                  (roomType === RoomType.QUICK
                    ? RoomType.QUICK + " _selected"
                    : RoomType.QUICK)
                }
                onClick={() =>
                  updateRoom({ ...room, roomType: RoomType.QUICK })
                }
              >
                <Typography variant="h6">{t("Quick game")}</Typography>
                <Typography variant="body2">
                  {t("Fird dice and other rules for quick match.")}
                </Typography>
              </Grid>
              <Grid
                item
                className={
                  "shuffle " +
                  (roomType === RoomType.SHUFFLE
                    ? RoomType.SHUFFLE + " _selected"
                    : RoomType.SHUFFLE)
                }
                onClick={() =>
                  updateRoom({ ...room, roomType: RoomType.SHUFFLE })
                }
              >
                <Typography variant="h6">{t("GMS Shuffle")}</Typography>
                <Typography variant="body2">
                  {t("Tournament for playing with different cards.")}
                </Typography>
              </Grid>
              <Grid
                item
                className={
                  "retro " +
                  (roomType === RoomType.RETRO
                    ? RoomType.RETRO + " _selected"
                    : RoomType.RETRO)
                }
                onClick={() =>
                  updateRoom({ ...room, roomType: RoomType.RETRO })
                }
              >
                <Typography variant="h6">{t("Retro")}</Typography>
                <Typography variant="body2">
                  {t("Rules for classic monopoly")}
                </Typography>
              </Grid>
              <Grid
                item
                className={
                  "roulette " +
                  (roomType === RoomType.ROULETTE
                    ? RoomType.ROULETTE + " _selected"
                    : RoomType.ROULETTE)
                }
                onClick={() =>
                  updateRoom({ ...room, roomType: RoomType.ROULETTE })
                }
              >
                <Typography variant="h6">{t("Russian roulette")}</Typography>
                <Typography variant="body2">
                  {t("Go and shoot. Nothing more.")}
                </Typography>
              </Grid>
            </Grid>
          </Grid>
          <Grid item sm={6} className="roomParams">
            <Grid container justify="space-between" alignItems="center">
              {roomType === RoomType.REGULAR && (
                <RegularGameParams isVip={!!user && user.vip} />
              )}
              {roomType === RoomType.QUICK && (
                <QuickGameParams isVip={!!user && user.vip} />
              )}
              {roomType === RoomType.RETRO && (
                <RetroGameParams isVip={!!user && user.vip} />
              )}
              {roomType === RoomType.ROULETTE && (
                <RouletteGameParams isVip={!!user && user.vip} />
              )}
              {roomType === RoomType.SHUFFLE && (
                <ShuffleGameParams isVip={!!user && user.vip} />
              )}
            </Grid>
          </Grid>
        </Grid>
      </DialogContent>
      <DialogActions>
        <Button
          className="_create-game-button"
          disabled={
            !user ||
            (!user.vip &&
              (roomType === RoomType.QUICK ||
                roomType === RoomType.RETRO ||
                roomType === RoomType.ROULETTE))
          }
          onClick={() => {
            createRoom();
            closeRoomModal();
          }}
          color="primary"
          variant="outlined"
        >
          {t("Create room")}
        </Button>
      </DialogActions>
    </Dialog>
  );
}
