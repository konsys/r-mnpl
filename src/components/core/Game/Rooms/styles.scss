@import "../../../../theme/styles/vars.scss";

/** findGames **/
.findGame {
  .newRoomOneParams {
    .regular {
      color: $regularСolor;
    }
    .quick {
      color: $quickСolor;
    }
    .shuffle {
      color: $shuffleСolor;
    }
    .retro {
      color: $retroСolor;
    }
    .roulette {
      color: $rouletteСolor;
    }
  }
  .avatar {
    width: 100px;
    height: 100px;
    div {
      width: 100%;
      height: 100%;
    }
    img {
      width: 100%;
      height: 100%;
    }
    svg {
      width: 50px;
      height: 50px;
      opacity: 0.4;
      cursor: pointer;
    }
    .avatarWrap {
      position: relative;
      &:hover {
        > svg {
          display: block;
        }
      }
      svg {
        display: none;
        position: absolute;
        right: 0px;
        top: 0px;
        z-index: 1000;
        width: 30px;
        height: 30px;
        cursor: pointer;
        color: $color-grapefruit-500;
      }
      img {
        border-radius: 50%;
        width: 80px;
        height: 80px;
      }
    }
    .addPlayerWrap {
      .noCursor {
        svg {
          cursor: auto !important;
        }
      }
      width: 50px;
      height: 50px;
      svg {
        width: 100%;
        height: 100%;
      }
    }
  }
}
