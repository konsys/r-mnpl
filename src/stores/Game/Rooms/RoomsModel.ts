import { GameDomain, user$ } from "../UserStore";
import { IResponceCode, IUser } from "types/types";
import {
  addPlayerToRoomFetch,
  createRoomFetch,
  fetchRooms,
  removePlayerFromRoomFetch,
} from "../../../api/Rooms/api";
import { closeGameModal, openGameModal } from "../GameModal/GameModalModel";
import { combine, sample } from "effector";

import { ErrorCode } from "utils/errors";
import { createGate } from "effector-react";
import nanoid from "nanoid";

export enum RoomPortalFieldType {
  PORTAL = "Portal",
  NOP = "Empty field",
  ROULETTE = "Roulette",
  RUSSIAN_ROULETTE = "Russian roulette",
}

export enum RoomType {
  REGULAR = "regular",
  RETRO = "retro",
  SHUFFLE = "shuffle",
  QUICK = "quick",
  ROULETTE = "roulette",
}
export enum RoomStatus {
  NOT_CREATED = "notCreated",
  PENDING = "pending",
  DELETED = "deleted",
  PLAYING = "playing",
  COMPLETED = "completed",
}
export enum RoomTypeName {
  REGULAR = "Regular game",
  RETRO = "Retro",
  SHUFFLE = "GMS Shuffle",
  QUICK = "Quick game",
  ROULETTE = "Russian roulette",
}

export enum PlayerRoomStatus {
  ACITVE = "active",
  SURRENDERED = "surrendered",
}

export interface RoomPlayer extends IUser {
  playerRoomStatus: PlayerRoomStatus;
}

export interface IRoomState {
  roomId: string;
  creatorId: number;
  players: Partial<RoomPlayer[]>;
  createTime: Date;
  roomType: RoomType;
  playersNumber: number;
  autostart: boolean;
  privateRoom: boolean;
  restarts: boolean;
  portalType: RoomPortalFieldType;
  roomStatus: RoomStatus;
}

export interface IAddPlayerToRoom {
  roomId: string;
  userId: number;
}

export interface IRoomResponce {
  playersInRooms: number;
  rooms: IRoomState[];
}

const RoomDomain = GameDomain.domain("ChatDomain");

export const roomsGate = createGate<any>();

export const setRooms = RoomDomain.event<IRoomResponce>();

export const createRoomFx = RoomDomain.effect<IRoomState, IResponceCode, Error>(
  {
    handler: createRoomFetch,
  }
);

createRoomFx.fail.watch((v: any) => {
  try {
    openGameModal({
      open: true,
      title: "Oops!",
      text:
        v.error.response.status === 401
          ? ErrorCode[v.error.response.status]
          : ErrorCode[v.error.response.data.code],
    });
  } catch (err) {
    closeGameModal();
  }
});

export const getRoomsFx = RoomDomain.effect<undefined, IRoomResponce, Error>({
  handler: fetchRooms,
});

export const addPlayerToRoomFx = RoomDomain.effect<
  IAddPlayerToRoom,
  IResponceCode,
  Error
>({
  handler: addPlayerToRoomFetch,
});

export const removePlayerFromRoomFx = RoomDomain.effect<
  IAddPlayerToRoom,
  IResponceCode,
  Error
>({
  handler: removePlayerFromRoomFetch,
});

addPlayerToRoomFx.fail.watch((v: any) => {
  try {
    openGameModal({
      open: true,
      title: "Oops!",
      text: ErrorCode[v.error.response.data.code] || "Unknown error",
    });
  } catch (err) {
    closeGameModal();
  }
});

export const createRoom = RoomDomain.event<void>();

export const updateRoom = RoomDomain.event<IRoomState>();
export const toggleAutostart = RoomDomain.event<void>();
export const togglePrivateRoom = RoomDomain.event<void>();
export const togglRestarts = RoomDomain.event<void>();
export const toggleRoomSwitch = RoomDomain.event<string>();
export const resetcurrentRoom$ = RoomDomain.event<void>();

export const currentRoom$ = RoomDomain.store<IRoomState>({
  roomId: "",
  creatorId: 0,
  players: [],
  createTime: new Date(),
  roomType: RoomType.REGULAR,
  playersNumber: 4,
  portalType: RoomPortalFieldType.ROULETTE,
  autostart: true,
  restarts: false,
  privateRoom: false,
  roomStatus: RoomStatus.NOT_CREATED,
})
  .on(updateRoom, (_, v) => v)

  .on(toggleAutostart, (state) => ({
    ...state,
    autostart: !state.autostart,
  }))
  .on(togglePrivateRoom, (state) => ({
    ...state,
    privateRoom: !state.privateRoom,
  }))
  .on(togglRestarts, (state) => ({
    ...state,
    restarts: !state.restarts,
  }))
  .reset(resetcurrentRoom$);

sample({
  clock: toggleRoomSwitch,
  source: toggleRoomSwitch,
  fn: (name) => {
    switch (name) {
      case "autostart":
        toggleAutostart();
        break;
      case "privateRoom":
        togglePrivateRoom();
        break;
      case "restarts":
        togglRestarts();
        break;
    }
  },
});

sample({
  clock: createRoom,
  source: combine({
    room: currentRoom$.map((v) => v),
    user: user$.map((v) => ({
      ...v,
      playerRoomStatus: PlayerRoomStatus.ACITVE,
    })),
  }),
  fn: ({ room, user }) => ({
    ...room,
    creatorId: user.userId,
    players: [user],
    roomId: `${nanoid(4)}-${Date.now()}`,
  }),

  target: createRoomFx,
});

export const resetRoomsStore = RoomDomain.event();
export const rooms$ = RoomDomain.store<IRoomResponce>({
  playersInRooms: 0,
  rooms: [],
})
  .reset(resetRoomsStore)
  .on(getRoomsFx.done, (_, { result }) => result)
  .on(setRooms, (_, result) => result);

export const myRooms$ = sample({
  clock: rooms$,
  source: combine({
    userId: user$.map((v) => v.userId),
    rooms: rooms$,
  }),
  fn: ({ userId, rooms }) => {
    const myRooms = rooms.rooms.filter(
      (r) =>
        (r.roomStatus === RoomStatus.PENDING ||
          r.roomStatus === RoomStatus.PLAYING) &&
        r.players.some((pl) => pl?.userId === userId)
    );
    return myRooms || [];
  },
});

rooms$.watch((v) => console.log("myRooms$", v));

export const playingRooms$ = sample({
  clock: rooms$,
  source: combine({
    rooms: rooms$,
  }),
  fn: ({ rooms }) => {
    const playing = rooms.rooms.filter(
      (r) => r.roomStatus === RoomStatus.PLAYING
    );
    return playing || [];
  },
});

sample({
  clock: roomsGate.open,
  source: roomsGate.state,
  target: getRoomsFx,
});
