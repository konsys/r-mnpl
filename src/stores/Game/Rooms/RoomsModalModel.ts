import { GameDomain } from "../UserModel";
import { resetRoomsStore } from "./RoomsModel";
import { sample } from "effector";

const RoomModalDomain = GameDomain.domain("GameModalDomain");

export const openRoomModal = RoomModalDomain.event();
export const closeRoomModal = RoomModalDomain.event();
export const roomModalStore = RoomModalDomain.store<boolean>(false)
  .on(openRoomModal, () => true)
  .reset(closeRoomModal);

sample({
  clock: closeRoomModal,
  source: resetRoomsStore,
  target: resetRoomsStore,
});
