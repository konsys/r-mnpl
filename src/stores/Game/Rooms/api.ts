import { IAddPlayerToRoom, IRoomResponce, IRoomState } from "./RoomsModel";

import { IResponceCode } from "types/types";
import { client } from "http/client";

const roomsUrl = `/rooms`;

export const fetchRooms = async (): Promise<IRoomResponce> => {
  return await (await client.get(roomsUrl)).data;
};

export const surrenderRoomFetch = async (): Promise<boolean> => {
  return await (await client.post(`${roomsUrl}/surrender`)).data;
};

export const createRoomFetch = async (
  params: IRoomState
): Promise<IResponceCode> => {
  return await (await client.post(roomsUrl, { room: params })).data;
};

export const addPlayerToRoomFetch = async (
  params?: IAddPlayerToRoom
): Promise<IResponceCode> => {
  return await (await client.post(`${roomsUrl}/addPlayer`, { add: params }))
    .data;
};

export const removePlayerFromRoomFetch = async (
  params?: IAddPlayerToRoom
): Promise<IResponceCode> => {
  return await (
    await client.post(`${roomsUrl}/removePlayer`, { remove: params })
  ).data;
};
