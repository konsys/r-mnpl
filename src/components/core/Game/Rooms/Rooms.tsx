import "./styles.scss";

import { Button, Divider, Grid, Typography } from "@material-ui/core";
import {
  PlayerRoomStatus,
  RoomStatus,
  myPendingRoom$,
  rooms$,
  roomsGate,
} from "stores/Game/Rooms/RoomsModel";
import { useGate, useStore } from "effector-react";

import { BLOCK_SPACING } from "../../../../theme";
import CreateRoomModal from "../CreateRoomModal/CreateRoomModal";
import React from "react";
import { Redirect } from "react-router-dom";
import RoomBlock from "./RoomBlock/RoomBlock";
import Template from "../../../views/Template/Template";
import { TemplateModules } from "types/types";
import { openRoomModal } from "stores/Game/Rooms/RoomsModalModel";
import { useTranslation } from "react-i18next";
import { user$ } from "stores/Game/User/UserModel";

export const Rooms = () => {
  useGate(roomsGate);
  const myPendingRoom = useStore(myPendingRoom$);
  const rooms = useStore(rooms$);
  const { t } = useTranslation();
  const user = useStore(user$);
  const userId = user && user.userId;

  const playingRooms = rooms.rooms.filter(
    (v) => v.roomStatus === RoomStatus.PLAYING
  );

  const myPlayingRoom = playingRooms.find((v) =>
    v.players.some(
      (v1) =>
        v1?.userId === userId && v1.playerRoomStatus === PlayerRoomStatus.ACITVE
    )
  );
  return (
    <>
      <CreateRoomModal />
      {myPlayingRoom ? (
        <Redirect
          to={{
            pathname: `/board/${myPlayingRoom.roomId}`,
          }}
        />
      ) : (
        <div className="findGame">
          <Template
            columns={2}
            title={"Search games"}
            modules={[
              TemplateModules.FRIENDS_ONLINE,
              TemplateModules.TOP_FIVE,
              // TemplateModules.BUY_GALLERY,
              TemplateModules.GAME_CHAT,
            ]}
          >
            <Grid
              container
              justify="space-between"
              spacing={BLOCK_SPACING}
              direction="row"
            >
              <Grid item>
                <Typography variant="h6">{t("Waiting for games")}</Typography>
              </Grid>

              <Grid item>
                <Typography variant="body2">
                  <Button
                    variant="outlined"
                    color="primary"
                    onClick={() => openRoomModal()}
                    disabled={!!myPendingRoom}
                  >
                    {t("Create room")}
                  </Button>
                </Typography>
              </Grid>
            </Grid>
            <Grid
              container
              justify="center"
              direction="column"
              spacing={BLOCK_SPACING}
            >
              <Grid item>
                <Divider />
              </Grid>
            </Grid>
            <Grid
              container
              justify="center"
              direction="column"
              spacing={BLOCK_SPACING}
            >
              {rooms &&
                Array.isArray(rooms.rooms) &&
                rooms.rooms.map(
                  (room, k) =>
                    room.roomStatus === RoomStatus.PENDING && (
                      <Grid item key={k}>
                        <RoomBlock
                          room={room}
                          iHaveRoom={!!myPendingRoom}
                          userId={userId || undefined}
                        />
                      </Grid>
                    )
                )}
            </Grid>
          </Template>
        </div>
      )}
    </>
  );
};
