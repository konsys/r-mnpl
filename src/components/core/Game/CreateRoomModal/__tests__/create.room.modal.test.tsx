import CreateRoomModal from "../CreateRoomModal";
import React from "react";
import { closeRoomModal } from "stores/Game/Rooms/RoomsModalModel";
import { shallow } from "enzyme";

describe("Create room modal test", () => {
  it("should render", () => {
    expect(shallow(<CreateRoomModal />)).toMatchSnapshot();
  });

  it("should render", () => {
    closeRoomModal();
    expect(shallow(<CreateRoomModal />).find(".newRoom")).toHaveLength(1);
  });
});
