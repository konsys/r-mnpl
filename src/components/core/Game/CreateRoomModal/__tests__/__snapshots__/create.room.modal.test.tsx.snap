// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create room modal test should render 1`] = `ShallowWrapper {}`;
